viewing_party/party.py

```diff
@@ -0,0 +1,254 @@
+            # add the movie title to a set of all the movies watched by the user's friends
+            movies_watched_all_friends.add(movie_title)
+
+    # return set of the movies watched by user's friends
+    return movies_watched_all_friends
+
+def get_unique_watched(user_data):
+
+    # get a list of the user's watched movies
+    users_movies = get_users_watched_movies(user_data)
+    # get a list of the user's friend's watched movies
+    friends_movies = get_friends_watched_movies(user_data)
+
+    # get the difference between the user's movies and the user's friend's movies
+    unique_movies = users_movies.difference(friends_movies)
+    
+    users_unique_movies = []
+    
+    # loop over the list of movies only the user has watched
+    for movie in unique_movies:
+        movie_title = {"title":movie}
+        # add the movie title to the list of unique movies
+        users_unique_movies.append(movie_title)
+
+    # return the list of movies only the user has watched
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    # get a list of the user's watched movies
+    users_movies = get_users_watched_movies(user_data)
+    # get a list of the user's friend's watched movies
+    friends_movies = get_friends_watched_movies(user_data)
+
+    # get the difference between the user's friend's movies and the user's movies
+    unique_movies = friends_movies.difference(users_movies)
+    
+    friends_unique_movies = []
+
+    # loop over the list of unique movies the user's friends have watched
+    for movie in unique_movies:
+        movie_title = {"title": movie}
+        # add the movie title to the list of unique movies
+        friends_unique_movies.append(movie_title)
+
+    # return the list of movies the user's friends have watched but the user has not
+    return friends_unique_movies
+
+def get_friends_unique_watched_with_details(user_data):
+
+    # get a list of movie only theuser's friends have watched
+    users_unwatched_movies = get_friends_unique_watched(user_data)
+    friends_list = user_data["friends"]
+
+    unwatched_movies_and_details = set()
+
+    # loop over the list of movies only the user's friends have watched
+    for unwatched_movie in users_unwatched_movies:
+        unwatched_movie_title = unwatched_movie["title"]
+        # loop over the list of the user's friends
+        for friend in friends_list:
+            friends_watched_list = friend["watched"]
+            # loop over the list of that friend's watched movies
+            for movie in friends_watched_list:
+                friends_movie_title = movie["title"]
+                # if a movie from user's unwatched movie list is in a friend's watched movies list
+                #  add the movie title and details (such as host or genre) to a list of user's unwatched movies
+                if unwatched_movie_title == friends_movie_title:
+                    movie_details = tuple(movie.values())
```

Comment:
If you want to try to go the set route, _and_ preserve additional data about the dictionary values, you gout make a tuple containing the key-value pairs from the dict with `tuple(movie.items())`, then  on the outside, you could convert them back into a dict with the dict() method (if it receives a collection of tuples, it unpacks them into key-value pairs). This is still a little dangerous (even as is with just the values) since if we imagine the hypothetical full movie structure, there might be additional mutable data (lists of actor information, etc) that would _not_ be allowed in a set.